### PR TITLE
chore(TS): Circle

### PR DESCRIPTION
**Pull request by ShaMan123**
## Description

Migrate shapes using the transformer

## Changes

- `SHARED_ATTRIBUTES` moved to parser
- `setRadius` not chainable

## Gist

View the diff without whitespace!

### CODE REVIEW

File: src/shapes/circle.class.ts [hunk 1]
```diff
@@ -1,232 +1,223 @@
...
+      const { radius } = this;
+      const start = degreesToRadians(this.startAngle),
+        end = degreesToRadians(this.endAngle),
+        startX = cos(start) * radius,
+        startY = sin(start) * radius,
+        endX = cos(end) * radius,
+        endY = sin(end) * radius,
+        largeFlag = angle > 180 ? '1' : '0';
+      return [
+        '<path d="M ' + startX + ' ' + startY,
```
Comment (asturur) [inline src/shapes/circle.class.ts:134]:
```suggestion
        `<path d="M ${startX} ${startY}`,
```

Comment (asturur) [reply src/shapes/circle.class.ts:134]: those we can change. Probably there will be a nicer way to get rid of those arrays in future too

File: src/shapes/circle.class.ts [hunk 2]
```diff
@@ -1,232 +1,223 @@
...
+   * @returns {Promise<Circle>}
+   */
+  static fromObject(object: object): Promise<Circle> {
+    return FabricObject._fromObject(Circle, object);
+  }
+}
+
+export const circleDefaultValues: Partial<TClassProperties<Circle>> = {
+  type: 'circle',
+  radius: 0,
```
Comment (asturur) [inline src/shapes/circle.class.ts:206]: should we bump radius to 100 as default? creating circles that are unusable by default seems just wrong

File: src/shapes/circle.class.ts [hunk 3]
```diff
@@ -1,232 +1,223 @@
...
+      const start = degreesToRadians(this.startAngle),
+        end = degreesToRadians(this.endAngle),
+        startX = cos(start) * radius,
+        startY = sin(start) * radius,
+        endX = cos(end) * radius,
+        endY = sin(end) * radius,
+        largeFlag = angle > 180 ? '1' : '0';
+      return [
+        '<path d="M ' + startX + ' ' + startY,
+        ' A ' + radius + ' ' + radius,
```
Comment (asturur) [inline src/shapes/circle.class.ts:135]:
```suggestion
        ` A ${radius} ${radius}`,
```

File: src/shapes/circle.class.ts [hunk 4]
```diff
@@ -1,232 +1,223 @@
...
+        ' 0 ',
+        +largeFlag + ' 1',
```
Comment (asturur) [inline src/shapes/circle.class.ts:137]:
```suggestion
        `${largeFlag} 1`,
```

File: src/shapes/circle.class.ts [hunk 5]
```diff
@@ -1,232 +1,223 @@
...
+        ' ' + endX + ' ' + endY,
```
Comment (asturur) [inline src/shapes/circle.class.ts:138]:
```suggestion
        ` ${endX} ${endY}`,
```

File: src/shapes/circle.class.ts (hunk 2 above)
Comment (ShaMan123) [reply src/shapes/circle.class.ts:206]: agreed

**Comment by asturur** [comment]
not bumping radius now to avoid breaking UTs, we can do next time. if it gets green i merge this

**Review by asturur — APPROVED** [review]

**Pull request merged**